File: Java/src/main/java/com/adaptionsoft/games/uglytrivia/CurrentPlayer.java
package com.adaptionsoft.games.uglytrivia;

public class CurrentPlayer {

    Players players;
    int currentPlayer = 0;

    public CurrentPlayer(Players players) {
        this.players = players;
    }

    public int get() {
        return currentPlayer;
    }

    public void inc() {
        currentPlayer = currentPlayer + 1;
        if (currentPlayer == players.count()) currentPlayer = 0;
    }

}

File: Csharp/Trivia/CurrentPlayer.cs
namespace UglyTrivia
{
    public class CurrentPlayer
    {
        internal Players players;

        internal int currentPlayer = 0;

        public CurrentPlayer(Players players)
        {
            this.players = players;
        }

        public int Get()
        {
            return this.currentPlayer;
        }

        public void Inc()
        {
            this.currentPlayer++;
            if (this.currentPlayer == this.players.Count()) this.currentPlayer = 0;
        }
    }
}
